frontend/src/views/ScheduleView.vue

```diff
@@ -1,109 +1,89 @@
-      :events="calendarEvents"
-      :schedules="schedulesPreviews"
-      @date-change="onDateChange"
-    />
-  </div>
-</template>
-
-<script setup>
-import { calendarViews } from '@/definitions';
+<script setup lang="ts">
+import { BookingCalendarView } from '@/definitions';
 import {
   ref, inject, computed, onMounted,
 } from 'vue';
 import { useI18n } from 'vue-i18n';
 import { useRoute } from 'vue-router';
 import { storeToRefs } from 'pinia';
-import ScheduleCreation from '@/components/ScheduleCreation';
+import { dayjsKey, callKey, refreshKey } from "@/keys";
+import {
+  RemoteEvent,
+  RemoteEventListResponse,
+  Schedule,
+  ScheduleListResponse,
+  ScheduleAppointment,
+  TimeFormatted,
+} from "@/models";
+import ScheduleCreation from '@/components/ScheduleCreation.vue';
+import CalendarQalendar from '@/components/CalendarQalendar.vue';
 
 // stores
 import { useAppointmentStore } from '@/stores/appointment-store';
 import { useCalendarStore } from '@/stores/calendar-store';
-import CalendarQalendar from '@/components/CalendarQalendar.vue';
-import { dayjsKey } from "@/keys";
 
 const { t } = useI18n();
 const route = useRoute();
 const dj = inject(dayjsKey);
-const call = inject('call');
-const refresh = inject('refresh');
+const call = inject(callKey);
+const refresh = inject(refreshKey);
 
 const appointmentStore = useAppointmentStore();
 const calendarStore = useCalendarStore();
 const { pendingAppointments } = storeToRefs(appointmentStore);
 const { connectedCalendars } = storeToRefs(calendarStore);
 
 // current selected date, if not in route: defaults to now
-const activeDate = ref(route.params.date ? dj(route.params.date) : dj());
+const activeDate = ref(route.params.date ? dj(route.params.date as string) : dj());
 const activeDateRange = ref({
   start: activeDate.value.startOf('month'),
   end: activeDate.value.endOf('month'),
 });
 
 // active menu item for tab navigation of calendar views
-const tabActive = ref(calendarViews.month);
+const tabActive = ref(BookingCalendarView.Month);
 
 // get remote calendar data for current year
-const calendarEvents = ref([]);
-const getRemoteEvents = async (from, to) => {
+const calendarEvents = ref<RemoteEvent[]>([]);
+const getRemoteEvents = async (from: string, to: string) => {
   // Most calendar impl are non-inclusive of the last day, so just add one to the day.
   const inclusiveTo = dj(to).add(1, 'day').format('YYYY-MM-DD');
 
   calendarEvents.value = [];
   await Promise.all(connectedCalendars.value.map(async (calendar) => {
-    const { data } = await call(`rmt/cal/${calendar.id}/${from}/${inclusiveTo}`).get().json();
+    const { data }: RemoteEventListResponse = await call(`rmt/cal/${calendar.id}/${from}/${inclusiveTo}`).get().json();
     if (Array.isArray(data.value)) {
-      calendarEvents.value.push(...data.value.map((e) => ({ ...e, duration: dj(e.end).diff(dj(e.start), 'minutes') })));
+      calendarEvents.value.push(
+        ...data.value.map((e) => ({
```

Comment:
I would suggest changing `e` for something that tells what it is. I see it's typed as `RemoteEvent`, maybe `event` would be nice